### PR TITLE
Add conditional migration for the automated corrections api_user

### PR DESCRIPTION
**Pull request by nullflux**
## Related Issue or Background Info

- @timothybest-usds  and I created this `api_user` by hand in production when programmatically correcting some bad data. This migration enshrines this user in code for consistency and history.

## Changes Proposed

- Add an `api_user` with the email address `corrections-noreply@simplereport.gov` if one does not exist

### UI
- [x] Any changes to the UI/UX are approved by design 
- [x] Any new or updated content (e.g. error messages) are approved by design 

### Testing
- [x] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [x] Database changes are submitted as a separate PR
  - [x] Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
  - [x] Any changes to tables that have custom no-PHI views are accompanied by changes to those views
- [x] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [x] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [x] Any dependencies introduced have been vetted and discussed

## Cloud
- [x] DevOps team has been notified if PR requires ops support
- [x] If there are changes that cannot be tested locally, this has been deployed to our Azure `test` environment for verification

### CODE REVIEW

**Comment by benwarfield-usds** [comment]
So, a thought occurred to me about these users, sometime recently when I wasn't at a keyboard: they don't exist in a meaningful sense in the test environment, except potentially for any tests that are run before the database is cleared for the first time. This is an inconsistency, though probably not a bad one—potentially worse in other cases where migrations can be run in an environment that can be wiped, though?

**Comment by nullflux** [comment]
> This is an inconsistency, though probably not a bad one—potentially worse in other cases where migrations can be run in an environment that can be wiped, though?

Yeah, I don't think this row _needs_ to exist in all of the environments. Happy to just close this PR w/o merging if @timothybest-usds agrees that it's unimportant

File: backend/src/main/resources/db/changelog/db.changelog-master.yaml [hunk 1]
```diff
@@ -2171,3 +2171,19 @@ databaseChangeLog:
...
+      id: add-automated-corrections-api_user
+      author: adam@skylight.digital
+      comment: Adds the automated corrections api_user to the database if they do not already exist
+      preConditions:
+        - onSqlOutput: FAIL
+        - onFail: MARK_RAN
+        - sqlCheck:
+            sql: |
+              SELECT case when count(1) = 0 then 0 else 1 end FROM ${database.defaultSchemaName}.api_user
+              WHERE login_email='corrections-noreply@simplereport.gov';
```
Comment (benwarfield-usds) [inline backend/src/main/resources/db/changelog/db.changelog-master.yaml:2184]:
```suggestion
              SELECT count(1) FROM ${database.defaultSchemaName}.api_user
              WHERE login_email='corrections-noreply@simplereport.gov';
```
Any reason not to just... get the count?

**Comment by benwarfield-usds** [comment]
If we do merge it, I would put the e-mail in a property, same as the migrations user. If we want to make it keep coming back, we could also turn it into an always-runs migration? Not sure of the value, though.

**Comment by nullflux** [comment]
Closing this as "unnecessary"

**Pull request closed without merging**